<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1011</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>601</width>
      <height>432</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertsListWidget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>111</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="halfEdgesListWidget">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>10</y>
      <width>111</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="facesListWidget">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>10</y>
      <width>111</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>280</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>280</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>280</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>370</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>370</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>370</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>370</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>390</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>390</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>390</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>440</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>440</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>420</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>420</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>420</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceRedSpinBox">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>420</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>440</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="splitEdgeButton">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>310</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Split</string>
    </property>
   </widget>
   <widget class="QPushButton" name="triangulateFaceButton">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>310</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivideButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>310</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QPushButton" name="skinMeshButton">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>560</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Skin</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="jointsTreeWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>291</width>
      <height>141</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">1</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>600</y>
      <width>261</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Joints</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateXButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>450</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate X</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateYButton">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>450</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotateZButton">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>450</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Z</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1011</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
    <addaction name="actionLoad_OBJ"/>
    <addaction name="actionLoad_JSON"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
  <action name="actionLoad_OBJ">
   <property name="text">
    <string>Load OBJ</string>
   </property>
  </action>
  <action name="actionLoad_JSON">
   <property name="text">
    <string>Load JSON</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
